FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -86,18 +61,21 @@ extension ModelDownloadTask {
     return "fbml_model__\(appName)__\(remoteModelInfo.name)"
   }
 
-  func download() {
+  func download(progressHandler: ProgressHandler?, completion: @escaping Completion) {
     downloader.downloadFile(with: remoteModelInfo.downloadURL,
                             progressHandler: { downloadedBytes, totalBytes in
                               /// Fraction of model file downloaded.
                               let calculatedProgress = Float(downloadedBytes) / Float(totalBytes)
-                              DispatchQueue.main.async {
-                                self.downloadHandlers.progressHandler?(calculatedProgress)
-                              }
+                              // TODO: Does this need to be on the main thread?
+                              progressHandler?(calculatedProgress)
```

Comment:
Do these handlers also need to be on the main thread, since they return to ModelDownloader for the final progress handling / completion?